torch/csrc/autograd/variable.h

```diff
@@ -31,8 +31,12 @@ struct Variable : public at::Tensor {
 
   // Implicitly casts a Tensor to a Variable. This should only be called on
   // Tensors which you know are actually Variables.
-  /*implicit*/ Variable(Tensor const & rhs) : Tensor(rhs) {}
-  /*implicit*/ Variable(Tensor && rhs) noexcept : Tensor(std::move(rhs)) {}
+  /*implicit*/ Variable(Tensor const & rhs) : Tensor(rhs) {
+    AT_ASSERT(rhs.isVariable(), "not variable");
```

Comment:
We also have `TORCH_ASSERTM` in `torch/csrc/assertions.h` which I think we've been using in the torch code.